platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java

```diff
@@ -190,4 +188,14 @@ public void shutdown() {
     PlatformMetricsRegistry.stop();
     LOGGER.info("Service - {} is shutdown.", getServiceName());
   }
+
+  private void shutdownWithError(Throwable exception) {
+    LOGGER.error("Shutting down due to unrecoverable exception", exception);
+    try {
+      this.shutdown();
+    } catch (Exception e) {
+      // Ignore if failed to shut down
```

Comment:
Can this happen?

Reply:
Potentially. The shutdown method calls various shutdown hooks, which each service is able to define itself. If any happen to throw (which seems like it's possible especially given it can now be called _before_ startup), we don't want them to actually prevent the shutdown from completing.

Reply:
Log or it would be too noisy?

Reply:
Not a noise volume concern, i was just concerned it would be misleading. Basically something has happened to cause the server to shutdown abnormally. If we hit more errors when trying to force that shutdown (from a server that may not have started in the first place), we don't want those to distract from the root cause.